guides/source/working_with_javascript_in_rails.md

```diff
@@ -3,561 +3,197 @@
+      format.turbo_stream { render turbo_stream: turbo_stream.prepend('posts', partial: 'post') }
+    else
+      format.html { render :new, status: :unprocessable_entity }
+    end
+  end
+end
 ```
 
-#### `data-type`
-
-It is also possible to define the Ajax `dataType` explicitly while performing
-requests for `data-remote` elements, by way of the `data-type` attribute.
+Finally, Turbo Streams can be initiated from a model or a background job using built-in helpers. These broadcasts can be used to update content via a WebSocket connection to all users, keeping page content fresh and bringing your application to life.
 
-### Confirmations
+To broadcast a Turbo Stream from a model combine a model callback like this:
 
-You can ask for an extra confirmation of the user by adding a `data-confirm`
-attribute on links and forms. The user will be presented with a JavaScript `confirm()`
-dialog containing the attribute's text. If the user chooses to cancel, the action
-doesn't take place.
+```ruby
+class Post < ApplicationRecord
+  after_create_commit { broadcast_append_to('posts') }
+end
+```
 
-Adding this attribute on links will trigger the dialog on click, and adding it
-on forms will trigger it on submit. For example:
+With a WebSocket connection set up on the page that should receive the updates like this:
 
 ```erb
-<%= link_to "Dangerous zone", dangerous_zone_path,
-  data: { confirm: 'Are you sure?' } %>
+<%= turbo_stream_from "posts" %>
 ```
 
-This generates:
+Learn more about Turbo Streams in the [Turbo Handbook](https://turbo.hotwired.dev/handbook/streams).
 
-```html
-<a href="..." data-confirm="Are you sure?">Dangerous zone</a>
-```
+Replacements for Rails/UJS functionality
+————————————
 
-The attribute is also allowed on form submit buttons. This allows you to customize
-the warning message depending on the button which was activated. In this case,
-you should **not** have `data-confirm` on the form itself.
+Rails UJS allowed developers to override the `method` of `<a>` tags to perform non-GET requests after a hyperlink click and to add confirmation dialogs before executing an action.
 
+Turbo provides built in replacements for both.
 
-### Automatic disabling
+### Method
 
-It is also possible to automatically disable an input while the form is submitting
-by using the `data-disable-with` attribute. This is to prevent accidental
-double-clicks from the user, which could result in duplicate HTTP requests that
-the backend may not detect as such. The value of the attribute is the text that will
-become the new value of the button in its disabled state.
+Activating hyperlinks always results in an HTTP GET request. However, if your application is [RESTful](https://en.wikipedia.org/wiki/Representational_State_Transfer), some links are in fact actions that change data on the server, and must be performed with non-GET requests. This attribute allows marking up such links with an explicit method such as “post”, “put” or “delete”.
```

Comment:
```suggestion
Clicking links always results in an HTTP GET request. If your application is [RESTful](https://en.wikipedia.org/wiki/Representational_State_Transfer), some links are in fact actions that change data on the server, and should be performed with non-GET requests. This attribute allows marking up such links with an explicit method such as “post”, “put” or “delete”.
```

Reply:
Maybe it's worth also mentioning `button_to`?

Reply:
Do you mean mentioning `button_to` as an alternative to using links with non-standard methods?

Reply:
Yeah, similar to how https://turbo.hotwired.dev/handbook/drive#performing-visits-with-a-different-method says

> You should also consider that for accessibility reasons, it’s better if use actual forms and buttons for anything that’s not a GET.